sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -96,6 +96,9 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
     case AlterNamespaceSetPropertiesStatement(NonSessionCatalog(catalog, nameParts), properties) =>
       AlterNamespaceSetProperties(catalog, nameParts, properties)
 
+    case AlterNamespaceSetLocationStatement(NonSessionCatalog(catalog, nameParts), location) =>
+      AlterNamespaceSetLocation(catalog, nameParts, location)
```

Comment:
Can't you reuse `AlterNamespaceSetProperties` if you set `LOCATION_TABLE_PROP`?

Reply:
+1 for @imback82 's comment.

Reply:
I tried reusing, but : 
1. V2SessionCatalog.LOCATION_TABLE_PROP can not be accessed from ResolveCatalogs. I think it's because V2SessionCatalog is in sql/core module and ResolveCatalogs is in sql/catalyst module. please correct me if i am wrong. And I don't want to hard code the "location" in ResolveCatalogs.
2. I'm not sure whether we should put the reusing logic of AlterNamespaceSetProperties in ResolveCatalogs, since i think ResolveCatalogs should be more about resolving catalog. And keeping AlterNamespaceSetLocation independent from AlterNamespaceSetProperties may be clearer.
@imback82 @dongjoon-hyun @cloud-fan WDYT?

Reply:
I see a similar logic here:
https://github.com/apache/spark/blob/c5f644c6ebb871d76f6b6b12b341ba761427492c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala#L81-L82
Also, I am not sure `V2SessionCatalog.LOCATION_TABLE_PROP` is the right constant you want to use even though the value is what you want. Maybe we need to have a set of constants for v2 catalogs.

Reply:
Yea, these special properties need to be recognized by implementations as well. I think we should define them as static variables in `Table` and `SupportsNamespaces`.

Reply:
BTW it's orthogonal to this PR. We can hardcode "location" for now.

Reply:
thank you all for review and suggestion, updated to reuse AlterNamespaceSetProperties and hard code "location" in ResolveCatalogs.